Tools/Scripts/libraries/webkitscmpy/webkitscmpy/local/git.py

```diff
@@ -888,18 +888,20 @@ def fetch(self, branch, remote='origin'):
 
     def pull(self, rebase=None, branch=None, remote='origin'):
         commit = self.commit() if self.is_svn or branch else None
-        code = run(
-            [self.executable(), 'pull'] + (
-                [remote, branch] if branch else []
-            ) + (
-                [] if rebase is None else ['--rebase={}'.format('True' if rebase else 'False')]
-            ), cwd=self.root_path,
-        ).returncode
-        if self.cache and rebase and branch != self.branch:
-            self.cache.clear(self.branch)
 
-        if not code and branch:
+        code = 0
+        if branch and self.branch != branch:
             code = self.fetch(branch=branch, remote=remote)
+        if code:
+            code = run(
+                [self.executable(), 'pull'] + (
+                    [remote, branch] if branch else []
+                ) + ['--autostash'] + (
+                    [] if rebase is None else ['--rebase={}'.format('True' if rebase else 'False')]
+                ), cwd=self.root_path,
```

Comment:
This is a very strange indentation style. We normally do it like this:
code = run([self.executable(), 'pull'] + ([remote, branch] if branch else [])
    + ['--autostash'] + ([] if rebase is None else ['--rebase={}'.format('True' if rebase else 'False')]),
    cwd=self.root_path,

Reply:
We use `pep 8` in Python, which is a little different that the styles elsewhere in the project.

That being said, this is pretty ugly as a one-liner no matter how you do it. Re-wrote it as something that's (hopefully) more readable.